### DISCUSSION TITLE
Auto-Generate collection from content file

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#2311] [Ideas]
When adding Tina to a site, creating the schema.ts can be tedious if a piece of content has many fields. I thought it would be handy to be able to generate a collection from an existing Markdown/JSON file

I made a little POC CLI tool that lets you create a collection from a document. Still some things to figure out (I'm currently just plopping it in a "/models" directory, so the dev still needs to import that collection into `schema.ts`).

https://www.loom.com/share/8638da102f90481696bd7aa2688234b3

- @logan-anderson pointed out that we can have an interactive prompt to select the file instead of typing it out (https://www.npmjs.com/package/inquirer-file-tree-selection-prompt) 
- The user may have to go in afterwards, and tweak a few fields (E.g, changing a text field to a reference).

Would people find this a valuable command if built into our "@tinacms/cli"?

**Comment by jeffsee55** [comment]
This has also been discussed a bit [here](https://github.com/tinacms/tinacms/discussions/2177)

**Comment by Perni1984** [comment]
@jamespohalloran I think every tool that helps to generate `schema.ts` automatically from other sources is valuable. We don't have general use case right now from *.md -> tina schema yet, but nevertheless I think it is a good addition to @tinacms/cli

**Comment by jamespohalloran** [comment]
Update on this! I spent a couple of hours and wired up a very quick and dirty POC, for how this would look with a UI:
https://schema-generator-beta.vercel.app/

Drag and drop in markdown files, and see the shape of their collections in your schema
<img width="1291" alt="Screen Shot 2022-02-01 at 10 53 20 AM" src="https://user-images.githubusercontent.com/3323181/151991655-3c760da1-93ad-4c01-ac25-4e3c83367050.png">

I think it could help the onboarding, specifically with [this section of the docs](https://tina.io/docs/schema/).
If others agree that it has value and we want to move forward with it, the POC could probably use some UX love, as well as adding a few little features (like prompting the user for a collection name/path).

**Reply by spbyrne** [reply]
This is awesome! Would love to pitch in on styling.

**Reply by jamespohalloran** [reply]
Awesome! I have the repo pushed up [here](https://github.com/tinacms/schema-generator) if you ever want to hack on it!
